### PR TITLE
Rename ActorSystem to ClusterSystem

### PR DESCRIPTION
**Pull request by yim-lee**
Part 1 of https://github.com/apple/swift-distributed-actors/issues/879

### CODE REVIEW

**Comment by yim-lee** [comment]
Rename of `ActorSystemSettings` will be done in a separate PR.

**Review by ktoso — APPROVED** [review]
LGTM, skimmed through it all and looks good!

There was quite a few small related types with old names still... Thank you a lot for spotting them all :)

File: Sources/DistributedActors/LifecycleMonitoring/LifecycleWatch.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ public extension LifecycleWatch {
     @discardableResult
     func watchTermination<Watchee>(
         of watchee: Watchee,
-        @_inheritActorContext @_implicitSelfCapture whenTerminated: @escaping @Sendable(ActorSystem.ActorID) -> Void,
+        @_inheritActorContext @_implicitSelfCapture whenTerminated: @escaping @Sendable(ID) -> Void,
```
Comment (ktoso) [inline Sources/DistributedActors/LifecycleMonitoring/LifecycleWatch.swift:29]: oh that looks silly... formatters don't understand that all so well I guess;

Technically `@Sendable (ID) -> Void` and not `@Sendable(ID) -> Void` but I don't mind, let's get the bigger PR in and we can always reformat later :)

**Comment by ktoso** [comment]
Awesome, everything LGTM!

**Pull request merged**